lib/matplotlib/axes/_subplots.py

```diff
@@ -163,6 +163,28 @@ def _make_twin_axes(self, *args, **kwargs):
         self._twinned_axes.join(self, twin)
         return twin
 
+    def __repr__(self):
+        label = f"label={self.get_label()!r}" if self.get_label() else ""
+        title = (
+            "title={"
+            + ",".join(
+                f"{k}:{self.get_title(loc=k)!r}"
+                for k in ["left", "center", "right"]
+                if self.get_title(loc=k)
+            )
+            + "}"
+            if any(self.get_title(loc=k) for k in ["left", "center", "right"])
+            else ""
+        )
+        xlabel = f"xlabel={self.get_xlabel()!r}" if self.get_xlabel() else ""
+        ylabel = f"ylabel={self.get_ylabel()!r}" if self.get_ylabel() else ""
+        return (
+            "<"
+            + f"{self.__class__.__name__}:"
+            + ", ".join(txt for txt in (title, xlabel, ylabel, label) if txt)
+            + ">"
+        )
```

Comment:
Honestly, I think this would be much simpler without all the nested comprehensions and ternaries.
```suggestion
    def __repr__(self):
        fields = []
        if self.get_label():
            fields += [f"label={self.get_label()!r}"]
        titles = []
        for k in ["left", "center", "right"]:
            title = self.get_title(loc=k)
            if title:
                titles.append(f"{k}:{title!r}")
        if titles:
            fields += ["title={" + ",".join(titles) + "}"]
        if self.get_xlabel():
            fields += [f"xlabel={self.get_xlabel()!r}"]
        if self.get_ylabel():
            fields += [f"ylabel={self.get_ylabel()!r}"]
        return f"<{self.__class__.__name__}:" + ", ".join(fields) + ">"
```
(using `+=` or `.append` as you see fit.)